client/src/Components/FilterSide/index.jsx

```diff
@@ -0,0 +1,181 @@
+import {
+  Divider,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  OutlinedInput,
+  Select,
+  TextField,
+  Switch,
+  Rating,
+  Slider,
+  Autocomplete,
+  FormGroup,
+} from "@mui/material";
+import React from "react";
+import "./style.css";
+import PropTypes from "prop-types";
+
+const ITEM_HEIGHT = 25;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
+const initilstate = {
+  name: "",
+  features: [],
+  city: "",
+  gender: "",
+  fulltime: false,
+  rating: 5,
+  price: [0, 150],
+};
+
+function FilterSide({ gymData }) {
+  const { cities, gymsFeatures, genders } = gymData;
+  const [inputfilter, setInputfilter] = React.useState(initilstate);
```

Comment:
What you are doing here that you share the same state for all your inputs, So when I type something  in name input I change the state for the all inputs I have, Which led to unnecessary rendering,
So try to create a separate state for all your inputs